src/Relude/Extra/Foldable1.hs

```diff
@@ -89,8 +89,7 @@ instance Foldable1 NonEmpty where
     {-# INLINE fold1 #-}
 
     foldMap1 :: Semigroup m => (a -> m) -> NonEmpty a -> m
-    foldMap1 f (a :| [])     = f a
-    foldMap1 f (a :| b : bs) = f a <> foldMap1 f (b :| bs)
+    foldMap1 f (a :| as) = foldr (\b g x -> f x <> g b) f as a
```

Comment:
Thanks for the improvement! I always appreciate the performance improvements! However, I find it a bit difficult to read code with `foldr` which uses a function as a zero value and a 3-argument lambda. Could we move this lambda into separate local `go` function under `where` with explicit type signature? If so, will the performance still be the same?

Reply:
Yep, we can do that, and the performance is still the same.